k9mail/src/main/java/com/fsck/k9/message/html/HttpUriParser.java

```diff
@@ -18,6 +16,8 @@
 class HttpUriParser implements UriParser {
     // This string represent character group sub-delim as described in RFC 3986
     private static final String SUB_DELIM = "!$&'()*+,;=";
+    private static final Pattern DOMAIN_PATTERN =
+            Pattern.compile("\\w([\\w-]*\\w)*(\\.\\w([\\w-]*\\w)*)*(:(\\d{0,5}))?");
```

Comment:
Unfortunately `\w` also includes the underscore which is not valid in host names.

You can use non-capturing groups by having `?:` as first characters inside the parentheses. That'll make it easier to later get to the content you do want to capture.

Example: https://github.com/itiboi/k-9/blob/cf9c3d078e6e7296f16d8d8a29905047eeeec36b/k9mail/src/main/java/com/fsck/k9/message/html/UriLinkifier.java#L17